src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java

```diff
@@ -577,6 +577,38 @@ public void insertBinaryVectorWithInvalidDecodedValue() {
               is(
                   "Bad binary vector value to shred: Invalid content in EJSON $binary wrapper: decoded value is not a multiple of 4 bytes long (3 bytes)"));
     }
+
+    @Test
+    public void insertBinaryVectorWithUnmatchedVectorDimension() {
```

Comment:
naming: maybe start with word "fail" (like "failToInsert....")?

I know existing tests aren't named that way but I think it is good for test method name to indicate that it is testing a failing insert.